test/WithdrawQueue.t.sol

```diff
@@ -739,20 +748,77 @@ contract WithdrawQueueTest is MainnetStarterTest, AdaptorHelperFunctions, ISolve
         address[] memory users = new address[](1);
         users[0] = userA;
         vm.startPrank(userA);
-        vm.expectRevert(
-            bytes(abi.encodeWithSelector(SimpleSolver.SimpleSolver___NotInSolveContextOrNotActiveSolver.selector))
-        );
+        vm.expectRevert(bytes(abi.encodeWithSelector(SimpleSolver.SimpleSolver___WrongInitiator.selector)));
         queue.solve(cellar, users, runData, address(simpleSolver));
         vm.stopPrank();
     }
 
-    function finishSolve(bytes calldata runData, uint256, uint256 assetApprovalAmount) external {
+    function testSimpleSolverReentrancy() external {
+        uint256 sharesToWithdraw = 1e6;
+        address attacker = vm.addr(0xA);
+        deal(address(this), attacker, sharesToWithdraw);
+
+        ERC4626 maliciousERC4626 = ERC4626(address(this));
+
+        // Attacker deposits into this malicious ERC4626, and joins queue.
+        vm.startPrank(attacker);
+        WithdrawQueue.WithdrawRequest memory req = WithdrawQueue.WithdrawRequest({
+            deadline: uint64(block.timestamp + 100),
+            inSolve: false,
+            executionSharePrice: 1e6, // attacker sets a ridiculous execution share price.
+            sharesToWithdraw: uint96(sharesToWithdraw)
+        });
+        ERC20(address(this)).safeApprove(address(queue), sharesToWithdraw);
+        queue.updateWithdrawRequest(maliciousERC4626, req);
+        vm.stopPrank();
+
+        address[] memory users = new address[](1);
+        users[0] = attacker;
+
+        vm.startPrank(attacker);
+
+        // Call p2pSolve on reentrancy.
+        simpleSolverFunctionToReenter = 1;
+        vm.expectRevert(bytes("REENTRANCY"));
+        simpleSolver.redeemSolve(maliciousERC4626, users, 0, type(uint256).max);
+
+        // Call redeemSolve on reentrancy.
```

Comment:
Should the called function be `p2pSolve` and `finishSolve` on the other two `expectRevert` tests here?

Reply:
No, so the way I am checking for reentrancy is the test contract has implemented a mock cellar interface, where the redeem function tries to reenter into the SimpleSolver(L817 in test file). But the only way to actually reenter is if the SimpleSolver calls the redeem function the test, which only happens if `redeemSolve` is used.

Reply:
We discussed this further. Thanks!